tests/integration/targets/ndo_physical_interface/tasks/main.yml

```diff
@@ -0,0 +1,502 @@
+  ansible.builtin.fail:
+    msg: 'Please define the following variables: mso_hostname, mso_username and mso_password.'
+  when: mso_hostname is not defined or mso_username is not defined or mso_password is not defined
+
+# CLEAN ENVIRONMENT
+- name: Set vars
+  ansible.builtin.set_fact:
+    mso_info: &mso_info
+      host: '{{ mso_hostname }}'
+      username: '{{ mso_username }}'
+      password: '{{ mso_password }}'
+      validate_certs: '{{ mso_validate_certs | default(false) }}'
+      use_ssl: '{{ mso_use_ssl | default(true) }}'
+      use_proxy: '{{ mso_use_proxy | default(true) }}'
+      output_level: '{{ mso_output_level | default("debug") }}'
+
+# QUERY VERSION
+- name: Query MSO version
+  cisco.mso.mso_version:
+    <<: *mso_info
+    state: query
+  register: version
+
+- name: Execute tasks only for MSO version > 4.3
+  when: version.current.version is version('4.3', '>=')
+  block:
+    - name: Remove fabric resource policy template
+      cisco.mso.ndo_template: &template_absent
+        <<: *mso_info
+        name: ansible_fabric_resource_policy_template
+        type: fabric_resource
+        state: absent
+
+    - name: Add fabric resource policy template
+      cisco.mso.ndo_template:
+        <<: *template_absent
+        state: present
+
+    # Create interface policy group in fabric policy template
+    - name: Remove fabric policy template
+      cisco.mso.ndo_template: &fabric_policy_template_absent
+        <<: *mso_info
+        name: ansible_fabric_policy_template
+        type: fabric_policy
+        state: absent
+
+    - name: Create a fabric policy template
+      cisco.mso.ndo_template:
+        <<: *fabric_policy_template_absent
+        state: present
+    
+    - name: Create interface policy group (physical policy)
+      cisco.mso.ndo_interface_setting: &add_interface_policy_group
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        name: ansible_interface_policy_group
+        interface_type: physical
+        state: present
+      register: add_interface_policy_group
+
+    - name: Create another interface policy group
+      cisco.mso.ndo_interface_setting:
+        <<: *add_interface_policy_group
+        name: ansible_interface_policy_group_2
+      register: add_interface_policy_group_2
+
+    # CREATE
+
+    # Create Physical Interface of type physical with physical policy UUID
+    - name: Create Physical Interface of type 'physical' (check mode)
+      cisco.mso.ndo_physical_interface: &add_physical_interface
+        <<: *mso_info
+        template: ansible_fabric_resource_policy_template
+        name: ansible_physical_interface
+        nodes: 103
+        interfaces: ['1/3']
+        physical_interface_type: physical
+        physical_policy_uuid: '{{ add_interface_policy_group.current.uuid }}'
+        state: present
+      check_mode: true
+      register: cm_physical_interface
+
+    - name: Create Physical Interface of type 'physical'
```

Comment:
CI appears broken, there are dangling objects for this policy without a template. This makes the testing fail.